numba_cuda/numba/cuda/tests/cudapy/test_inspect.py

```diff
@@ -108,6 +115,55 @@ def _test_inspect_sass(self, kernel, name, sass):
         self.assertIn("BRA", sass)  # Branch
         self.assertIn("EXIT", sass)  # Exit program
 
+    @skip_on_cudasim("Simulator does not generate code to be inspected")
+    @skip_if_nvjitlink_missing("nvJitLink is required for LTO")
+    def test_inspect_lto_asm(self):
+        ffi = cffi.FFI()
+        temp_add_cu = tempfile.NamedTemporaryFile(suffix=".cu")
+
+        with open(temp_add_cu.name, "w") as f:
+            f.write("""
+            #include <cuda_fp16.h>
+            extern "C"
+            __device__ int add_f2_f2(__half * res, __half * a, __half *b) {
+                *res = *a + *b;
+                return 0;
+            }
+            """)
+
+        add = declare_device(
+            "add_f2_f2",
+            float16(CPointer(float16), CPointer(float16)),
+            link=temp_add_cu.name,
+        )
+
+        @cuda.jit
+        def k(arr):
+            local_arr = cuda.local.array(shape=1, dtype=np.float16)
+            local_arr2 = cuda.local.array(shape=1, dtype=np.float16)
+            local_arr[0] = 1
+            local_arr2[0] = 2
+
+            ptr = ffi.from_buffer(local_arr)
+            ptr2 = ffi.from_buffer(local_arr2)
+
+            arr[0] = add(ptr, ptr2)
+
+        arr = np.array([0], dtype=np.float16)
+
+        k[1, 1](arr)
+
+        allasms = k.inspect_asm()
+        asm = next(iter(allasms.values()))
+
+        regex = re.compile(r"call(.|\n)*add_f2_f2")
+        self.assertRegex(asm, regex)
+
+        all_ext_asms = k.inspect_lto_ptx()
+        lto_asm = next(iter(all_ext_asms.values()))
+
+        assert "add.f16" in str(lto_asm)
```

Comment:
The assertions usually follow this form: 

```suggestion
        self.assertIn("add.f16", str(lto_asm))
```

(I think the pattern in the PR is more normal with pytest?)

Reply:
Done in [7798108](https://github.com/NVIDIA/numba-cuda/pull/326/commits/7798108f106ccff61f0f9a189feda9af1bfb2844), [04eb522](https://github.com/NVIDIA/numba-cuda/pull/326/commits/04eb5228a387f1bd1690ed461dc16a21b5a74b36)